src/main/java/org/openbaton/drivers/openstack4j/OpenStack4JDriver.java

```diff
@@ -472,36 +482,68 @@ private String getExternalNetworkId(OSClient os, String internalNetworkName) thr
     }
 
     if (null == routerPort) {
-      throw new Exception("cannot find a connection to a router, cannot assign floating ip");
+      throw new Exception(
+          "Cannot find a connection to a router, therefore cannot assign floating ip");
     }
 
     // major ASSUMPTION:  There will only be ONE router connected to a given internal network
     Router router = os.networking().router().get(routerPort.getDeviceId());
-    log.debug("router is " + router);
-
+    log.debug(
+        "The internal network "
+            + internalNetworkId
+            + " is connected to the router "
+            + new GsonBuilder().setPrettyPrinting().create().toJson(router));
     return router.getExternalGatewayInfo().getNetworkId();
   }
 
-  private List<NetFloatingIP> listFloatingIps(OSClient os, String tenantId, String networkName) {
```

Comment:
Needs to make sure that the floating ip is associated with a pool that is available to the internalNetworkName when it is specified.